compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -457,24 +458,23 @@ execLint inputFile opts =
          DlintDisabled  -> opts{optDlintUsage=DlintEnabled defaultDir True}
 
 -- | Parse the daml.yaml for package specific config fields.
-parseProjectConfig :: ProjectConfig -> Either ConfigError PackageConfigFields
-parseProjectConfig project = do
-    name <- queryProjectConfigRequired ["name"] project
-    main <- queryProjectConfigRequired ["source"] project
-    exposedModules <- queryProjectConfig ["exposed-modules"] project
-    version <- queryProjectConfigRequired ["version"] project
-    dependencies <-
-        queryProjectConfigRequired ["dependencies"] project
-    dataDeps <- fromMaybe [] <$> queryProjectConfig ["data-dependencies"] project
-    sdkVersion <- queryProjectConfigRequired ["sdk-version"] project
-    cliOpts <- queryProjectConfig ["build-options"] project
-    Right $ PackageConfigFields name main exposedModules version dependencies dataDeps sdkVersion cliOpts
+parseProjectConfig :: Maybe PackageSdkVersion -> ProjectConfig -> Either ConfigError PackageConfigFields
+parseProjectConfig sdkVersionM project = do
+    pName <- queryProjectConfigRequired ["name"] project
+    pSrc <- queryProjectConfigRequired ["source"] project
+    pExposedModules <- queryProjectConfig ["exposed-modules"] project
+    pVersion <- queryProjectConfigRequired ["version"] project
+    pDependencies <- queryProjectConfigRequired ["dependencies"] project
+    pDataDependencies <- fromMaybe [] <$> queryProjectConfig ["data-dependencies"] project
+    pSdkVersion <- maybe (queryProjectConfigRequired ["sdk-version"] project) Right sdkVersionM
```

Comment:
This looks like `sdk-version` is now no longer required if the environment variable is set. Is this intended? I think it’s probably better to always enforce it to exist and maybe emit a warning if it is being overwritten by the environment variable.

Reply:
That sounds like a better idea. I'll make it so.